actionpack/lib/action_dispatch/http/mime_negotiation.rb

```diff
@@ -75,10 +75,12 @@ def formats
 
       # Sets the \variant for template.
       def variant=(variant)
-        if variant.is_a?(Symbol)
-          @variant = [variant]
-        elsif variant.nil? || variant.is_a?(Array) && variant.any? && variant.all?{ |v| v.is_a?(Symbol) }
+        if variant.nil?
           @variant = variant
```

Comment:
Ah, `Array(nil)` returns an empty array, so there's no need for the compact call.